extensions/microprofile-metrics/deployment/src/main/java/org/apache/camel/quarkus/component/microprofile/metrics/deployment/MicroProfileMetricsProcessor.java

```diff
@@ -41,7 +42,7 @@ FeatureBuildItem feature() {
     CamelBeanBuildItem metricRegistry(CamelMicroProfileMetricsRecorder recorder) {
         return new CamelBeanBuildItem(
                 MicroProfileMetricsConstants.METRIC_REGISTRY_NAME,
-                MetricRegistry.class,
+                MetricRegistry.class.getName(),
```

Comment:
This should also be a DotName.

Reply:
If read the PR comment, you'd find that I will follow up this PR with a new one in which I'll review how the bean build items are used by extensions.

Reply:
Sorry, I was not sure you mean exactly this.